src/cli/deploy.rs

```diff
@@ -115,15 +115,24 @@ pub async fn run(deploy_args: DeployArgs) -> exitcode::ExitCode {
         }
     };
 
-    let local_replicas = validated_config.scaling.desired_replicas;
+    let local_replicas = validated_config
+        .scaling
+        .as_ref()
+        .and_then(|local_scaling_config| local_scaling_config.desired_replicas);
 
     // Warn if local scaling config differs from remote
     let has_scaling_config_drift = cage_scaling_config
         .as_ref()
-        .is_some_and(|config| config.desired_replicas() != local_replicas);
-    if has_scaling_config_drift {
+        .and_then(|config| local_replicas.map(|replicas| config.desired_replicas() != replicas))
+        .unwrap_or(true);
```

Comment:
Should this be `unwrap_or(false)`? (or `is_some_and`)

If either remote or local scaling config isn't set, then there'll be no drift